lighthouse-core/gather/gather-runner.js

```diff
@@ -197,6 +200,11 @@ class GatherRunner {
                 // Merge pass trace and network data into tracingData.
                 config.trace && Object.assign(tracingData.traces, loadData.traces);
                 config.network && (tracingData.networkRecords = loadData.networkRecords);
+              })
+              .then(_ => {
+                // Update the outer options object of any post-redirect url
+                // Subsequent passes will use the new URL, and it'll be reported outwards
+                options.url = runOptions.url;
```

Comment:
Did it mostly to save time in our later passes, but I agree it changes behavior. I'm fine with reusing the same initialUrl for each pass